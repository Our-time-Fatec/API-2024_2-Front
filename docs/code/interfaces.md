
---
title: interfaces
description: 'Pasta que contém as definições de interfaces utilizadas no projeto.'
---
# interfaces

A pasta `interfaces` contém as definições de interfaces que são utilizadas em todo o projeto. As interfaces são fundamentais para garantir a tipagem e a estrutura dos dados, proporcionando uma melhor organização e manutenção do código.

## Estrutura da Pasta

Abaixo estão as interfaces disponíveis nesta pasta:

- **IAlimento.ts**: Define a estrutura de um alimento, incluindo suas propriedades e tipos.
- **ICategoria.ts**: Define a estrutura de uma categoria de alimentos.
- **IDieta.ts**: Define a estrutura de uma dieta, incluindo os alimentos e categorias associadas.
- **IGrafico.ts**: Define a estrutura de dados para gráficos utilizados no aplicativo.
- **ILogin.ts**: Define a estrutura de dados para o processo de login.
- **IUsuario.ts**: Define a estrutura de um usuário, incluindo suas informações e preferências.

Essas interfaces são utilizadas em diversos componentes e serviços do projeto, garantindo a consistência e a segurança dos dados manipulados.
