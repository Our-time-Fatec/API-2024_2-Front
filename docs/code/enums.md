
---
title: enums
description: 'Contém definições de enums utilizados na aplicação.'
---
# Enums

A pasta `enums` contém definições de enums que são utilizados em diferentes partes da aplicação. Os enums são uma forma de definir um conjunto de constantes nomeadas, que podem ser utilizadas para melhorar a legibilidade e a manutenção do código.

## Estrutura

A estrutura da pasta `enums` é a seguinte:

- `diasSemana.ts`: Enum que representa os dias da semana.

## Uso

Os enums definidos nesta pasta podem ser importados e utilizados em outros arquivos da aplicação para garantir que apenas valores válidos sejam utilizados, evitando erros e melhorando a clareza do código.
