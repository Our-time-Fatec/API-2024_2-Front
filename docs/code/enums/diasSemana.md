---
title: diasSemana
description: 'Enumeração que representa os dias da semana.'
---

# DiasSemana

A enumeração `DiasSemana` define os dias da semana com seus respectivos nomes. Esta enumeração pode ser utilizada em diferentes partes do código para garantir a consistência ao referenciar os dias da semana.

## Enumeração

```typescript
export enum DiasSemana {
    Domingo = "Domingo",
    Segunda = "Segunda-feira",
    Terca = "Terça-feira",
    Quarta = "Quarta-feira",
    Quinta = "Quinta-feira",
    Sexta = "Sexta-feira",
    Sabado = "Sábado",
}
```

### Valores

- **Domingo**: Representa o primeiro dia da semana.
- **Segunda-feira**: Representa o segundo dia da semana.
- **Terça-feira**: Representa o terceiro dia da semana.
- **Quarta-feira**: Representa o quarto dia da semana.
- **Quinta-feira**: Representa o quinto dia da semana.
- **Sexta-feira**: Representa o sexto dia da semana.
- **Sábado**: Representa o sétimo dia da semana.

## Uso

A enumeração `DiasSemana` pode ser utilizada em condições, loops e outras estruturas de controle para facilitar a manipulação e a verificação dos dias da semana em aplicações.