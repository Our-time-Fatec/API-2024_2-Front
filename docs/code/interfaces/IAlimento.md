---
title: IAlimento
description: 'Interface que define a estrutura de um alimento, incluindo suas propriedades nutricionais e informações adicionais.'
---

# IAlimento

A interface `IAlimento` descreve a estrutura de um alimento, incluindo suas propriedades nutricionais e informações adicionais. Esta interface é utilizada para garantir que os objetos de alimento sigam um formato consistente em toda a aplicação.

## Estrutura da Interface

### IAlimento

```typescript
interface IAlimento {
    _id?: string;                // Identificador único do alimento (opcional)
    nome: string;                // Nome do alimento
    preparo: string;             // Modo de preparo do alimento
    categoriaUrl?: string;       // URL da categoria do alimento (opcional)
    porcao: number;              // Tamanho da porção em gramas
    quantidade?: number;         // Quantidade disponível (opcional)
    categoriaNome?: string;      // Nome da categoria do alimento (opcional)
    categoriaCodigo: number;     // Código da categoria do alimento
    detalhes: Detalhes;          // Detalhes nutricionais do alimento
    criadoEm?: Date;             // Data de criação do alimento (opcional)
}
```

### Detalhes

```typescript
interface Detalhes {
    valorEnergetico: number;     // Valor energético em kcal
    proteinas: number;           // Quantidade de proteínas em gramas
    carboidratos: number;        // Quantidade de carboidratos em gramas
    fibras: number;              // Quantidade de fibras em gramas
    lipidios: number;            // Quantidade de lipídios em gramas
}
```

## Exportação

A interface `IAlimento` e a interface `Detalhes` são exportadas para serem utilizadas em outras partes da aplicação.

```typescript
export { IAlimento, Detalhes };
```