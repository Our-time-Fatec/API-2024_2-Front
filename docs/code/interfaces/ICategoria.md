---
title: ICategoria
description: 'Interface que define a estrutura de uma categoria no sistema.'
---

# ICategoria

A interface `ICategoria` define a estrutura de uma categoria no sistema, incluindo suas propriedades essenciais.

## Propriedades

- **_id**: `string`  
  Identificador único da categoria.

- **nome**: `string`  
  Nome da categoria.

- **codigo**: `number`  
  Código numérico associado à categoria.

- **urlPlaceholder**: `string`  
  URL de um placeholder relacionado à categoria.

## Exportação

A interface é exportada como padrão para ser utilizada em outras partes do sistema:

```typescript
export default ICategoria;
```