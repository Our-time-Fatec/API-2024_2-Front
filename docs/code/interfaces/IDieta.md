---
title: IDieta
description: 'Interfaces relacionadas à dieta, incluindo detalhes, grupos e consumo de alimentos.'
---

# IDieta

Este arquivo contém definições de interfaces relacionadas à estrutura de dados de dietas, incluindo detalhes, grupos de alimentos e consumo diário.

## Interfaces

### IDietaDetalhes
```typescript
export interface IDietaDetalhes extends Detalhes {
}
```
Interface que estende a interface `Detalhes`, representando detalhes adicionais de uma dieta.

### IAlimentoDieta
```typescript
export interface IAlimentoDieta {
    alimentoId: string;
    nome?: string;
    preparo?: string;
    porcao: number;
    quantidade: number;
    categoriaCodigo?: number;
    detalhes?: Detalhes;
}
```
Interface que representa um alimento dentro de uma dieta. Contém informações como ID, nome, preparo, porção, quantidade e categoria.

### IGrupo
```typescript
export interface IGrupo {
    _id?: string;
    nome: string;
    alimentos: IAlimentoDieta[];
}
```
Interface que representa um grupo de alimentos. Inclui um ID opcional, nome do grupo e uma lista de alimentos (`IAlimentoDieta`).

### IDietaFixa
```typescript
export interface IDietaFixa {
    _id?: string;
    usuarioId?: string;
    diaSemana: DiasSemana[] | DiasSemana;
    criadoEm?: Date;
    atualizadoEm?: Date | null;
    removidoEm?: Date | null;
    detalhes?: IDietaDetalhes;
    grupos: IGrupo[];
}
```
Interface que representa uma dieta fixa. Inclui informações sobre o usuário, dias da semana, datas de criação, atualização e remoção, detalhes da dieta e grupos de alimentos.

### IDietaDiaria
```typescript
export interface IDietaDiaria {
    usuarioId: string;
    diaSemana: DiasSemana;
    dia: Date;
    gruposConsumo: IGrupoConsumo[];
}
```
Interface que representa a dieta diária de um usuário, incluindo o ID do usuário, dia da semana, data e grupos de consumo.

### IGrupoConsumo
```typescript
export interface IGrupoConsumo {
    _id: string;
    grupo: string;  
    alimentos: IContagem[]; 
}
```
Interface que representa um grupo de consumo, contendo um ID, nome do grupo e uma lista de contagens de alimentos.

### IContagem
```typescript
export interface IContagem {
    consumido: number;
    paraConsumir: number;
    alimento: IAlimentoConsumo; // Referência à interface IAlimento
}
```
Interface que representa a contagem de um alimento consumido e a quantidade a ser consumida, referenciando a interface `IAlimentoConsumo`.

### IAlimentoConsumo
```typescript
interface IAlimentoConsumo {
    _id: string,
    alimentoId: string;
    nome: string;
    preparo: string;
    categoriaUrl?: string;
    porcao: number;
    quantidade?: number;
    categoriaNome?: string;
    categoriaCodigo: number;
    detalhes: Detalhes;
    criadoEm?: Date;
    nomeGrupo?: string,
}
```
Interface que representa um alimento consumido, incluindo informações como ID, nome, preparo, categoria, porção e detalhes adicionais.