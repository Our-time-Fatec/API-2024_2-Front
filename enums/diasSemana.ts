export enum DiasSemana {
    Domingo = "Domingo",
    Segunda = "Segunda-feira",
    Terca = "Terça-feira",
    Quarta = "Quarta-feira",
    Quinta = "Quinta-feira",
    Sexta = "Sexta-feira",
    Sabado = "Sábado",
}