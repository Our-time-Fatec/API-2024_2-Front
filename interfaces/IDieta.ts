    import { DiasSemana } from "../enums/diasSemana";
    import { Detalhes } from "./IAlimento";

    export interface IDietaDetalhes extends Detalhes {
    }

    export interface IAlimentoDieta {
        alimentoId: string;
        nome?: string;
        preparo?: string;
        porcao: number;
        quantidade: number;
        categoriaCodigo: Number;
        detalhes: Detalhes;
    }

    export interface IAlimentoGrupo{
        alimentoId: string;
        porcao: number;
        quantidade: number
    }

    export interface IGrupo {
        _id?: string;
        nome: string;
        alimentos: IAlimentoDieta[];
    }

    export interface IDietaFixa {
        _id?: string;
        usuarioId?: string;
        diaSemana: DiasSemana[] | DiasSemana;
        criadoEm?: Date;
        atualizadoEm?: Date | null;
        removidoEm?: Date | null;
        detalhes?: IDietaDetalhes;
        grupos: IGrupo[];
    }