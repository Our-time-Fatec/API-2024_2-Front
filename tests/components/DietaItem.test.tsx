import React from 'react';
import renderer, { act } from 'react-test-renderer';
import DietaItem from '../../components/dieta'; // ajuste o caminho conforme necessário
import { IDietaFixa } from '../../interfaces/IDieta'; // ajuste o caminho conforme necessário
import { DiasSemana } from '../../enums/diasSemana';
import { Modal, TouchableOpacity } from 'react-native';

// Mock para as props de dieta
const dietaMock: IDietaFixa = {
  _id: '1',
  diaSemana: DiasSemana.Segunda,
  criadoEm: new Date(),
  detalhes: {
    valorEnergetico: 2000,
    proteinas: 50,
    carboidratos: 300,
    fibras: 10,
    lipidios: 20,
  },
  grupos: [
    {
      _id: 'grupo1',
      nome: 'Grupo 1',
      alimentos: [
        {
          alimentoId: 'alimento1',
          nome: 'Alimento 1',
          quantidade: 2,
          porcao: 100,
          detalhes: {
              valorEnergetico: 100,
              proteinas: 0,
              carboidratos: 0,
              fibras: 0,
              lipidios: 0
          },
        },
      ],
    },
  ],
};

describe('Componente DietaItem', () => {
  it('deve renderizar corretamente', () => {
    const tree = renderer.create(
      <DietaItem
        dieta={dietaMock}
        isUserDieta={true}
        onEdit={jest.fn()}
        onDelete={jest.fn()}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot(); // Verifica se a renderização corresponde ao snapshot
  });

  it('deve abrir o modal ao pressionar o botão', () => {
    const component = renderer.create(
      <DietaItem
        dieta={dietaMock}
        isUserDieta={true}
        onEdit={jest.fn()}
        onDelete={jest.fn()}
      />
    );

    // Pressiona o TouchableOpacity para abrir o modal
    act(() => {
      component.root.findByProps({ testID: "open-modal-button" }).props.onPress();
    });

    // Verifica se o modal está visível
    const modal = component.root.findByType(Modal);
    expect(modal.props.visible).toBe(true);
  });

  it('deve chamar a função onEdit ao pressionar o botão Editar', () => {
    const onEditMock = jest.fn();
    const component = renderer.create(
      <DietaItem
        dieta={dietaMock}
        isUserDieta={true}
        onEdit={onEditMock}
        onDelete={jest.fn()}
      />
    );

    // Abre o modal antes de pressionar o botão Editar
    act(() => {
      component.root.findByProps({ testID: "open-modal-button" }).props.onPress();
    });

    // Pressiona o botão Editar
    act(() => {
      component.root.findByProps({ testID: "edit-button" }).props.onPress();
    });

    expect(onEditMock).toHaveBeenCalledWith('1'); // Verifica se a função foi chamada com o ID correto
  });

  it('deve chamar a função onDelete ao pressionar o botão Remover', () => {
    const onDeleteMock = jest.fn();
    const component = renderer.create(
      <DietaItem
        dieta={dietaMock}
        isUserDieta={true}
        onEdit={jest.fn()}
        onDelete={onDeleteMock}
      />
    );

    // Abre o modal antes de pressionar o botão Remover
    act(() => {
      component.root.findByProps({ testID: "open-modal-button" }).props.onPress();
    });

    // Pressiona o botão Remover
    act(() => {
      component.root.findByProps({ testID: "delete-button" }).props.onPress();
    });

    expect(onDeleteMock).toHaveBeenCalledWith('1'); // Verifica se a função foi chamada com o ID correto
  });

  it('deve fechar o modal ao pressionar o botão Fechar', () => {
    const component = renderer.create(
      <DietaItem
        dieta={dietaMock}
        isUserDieta={true}
        onEdit={jest.fn()}
        onDelete={jest.fn()}
      />
    );

    // Abre o modal antes de pressionar o botão Fechar
    act(() => {
      component.root.findByProps({ testID: "open-modal-button" }).props.onPress();
    });

    // Pressiona o botão Fechar
    act(() => {
      component.root.findByProps({ testID: "close-modal-button" }).props.onPress();
    });

    // Verifica se o modal está fechado
    const modal = component.root.findByType(Modal);
    expect(modal.props.visible).toBe(false);
  });
});
