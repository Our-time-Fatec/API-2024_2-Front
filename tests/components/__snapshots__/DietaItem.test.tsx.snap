// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente DietaItem deve renderizar corretamente 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#ddd",
      "borderRadius": 8,
      "borderWidth": 1,
      "elevation": 2,
      "flex": 1,
      "margin": 8,
      "overflow": "hidden",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="open-modal-button"
  >
    <View
      style={
        {
          "padding": 8,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "bold",
            "marginBottom": 4,
          }
        }
      >
        Dia: 
        Segunda-feira
      </Text>
      <Text
        style={
          {
            "color": "#555",
            "fontSize": 14,
            "marginBottom": 2,
          }
        }
      >
        Calorias:
         
        2000.00
         
        kcal
      </Text>
      <Text
        style={
          {
            "color": "#555",
            "fontSize": 14,
            "marginBottom": 2,
          }
        }
      >
        Data de criação:
         
        30/10/2024
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 3,
        "justifyContent": "space-between",
        "marginBottom": 10,
        "paddingHorizontal": 10,
        "paddingVertical": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2d74da",
          "borderRadius": 5,
          "flex": 0.5,
          "marginTop": 5,
          "opacity": 1,
          "paddingVertical": 8,
        }
      }
      testID="edit-button"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
          }
        }
      >
        Editar
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e13f2f",
          "borderRadius": 5,
          "flex": 0.5,
          "marginTop": 5,
          "opacity": 1,
          "paddingVertical": 8,
        }
      }
      testID="delete-button"
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
          }
        }
      >
        Remover
      </Text>
    </View>
  </View>
</View>
`;
